agent-install/agent-install.sh

```diff
@@ -548,12 +549,21 @@ function get_all_variables() {
     ARCH=$(get_arch)
     log_info "OS: $OS, Distro: $DISTRO, Distro Release: $DISTRO_VERSION_NUM, Distro Code Name: $CODENAME, Architecture: $ARCH"
 
-    # The edge node id can be specified 3 different ways: -d (NODE_ID), the first part of -a (HZN_EXCHANGE_NODE_AUTH), or HZN_DEVICE_ID. Need to reconcile all of them.
+    # The edge node id can be specified 4 different ways: -d (HZN_NODE_ID), the first part of -a (HZN_EXCHANGE_NODE_AUTH), NODE_ID(deprecated) or HZN_DEVICE_ID. Need to reconcile all of them.
     local node_id   # just used in this section of code to sort out this mess
     # find the 1st occurrence of the user specifying node it
-    if [[ -n ${HZN_EXCHANGE_NODE_AUTH%%:*} ]]; then node_id=${HZN_EXCHANGE_NODE_AUTH%%:*}
-    elif [[ -n $NODE_ID ]]; then node_id=$NODE_ID
-    elif [[ -n $HZN_DEVICE_ID ]]; then node_id=$HZN_DEVICE_ID
+    if [[ -n ${HZN_EXCHANGE_NODE_AUTH%%:*} ]]; then 
+        node_id=${HZN_EXCHANGE_NODE_AUTH%%:*}
+        log_info "Using node id from HZN_EXCHANGE_NODE_AUTH"
+    elif [[ -n $HZN_NODE_ID ]]; then 
+        node_id=$HZN_NODE_ID
+        log_info "Using node id from HZN_NODE_ID"
+    elif [[ -n $NODE_ID ]]; then 
+        node_id=$NODE_ID
+        log_warning "Using node id from NODE_ID. NODE_ID is deprecated, please use HZN_NODE_ID"
```

Comment:
Deprecated means it still works now, but won't sometime in the future. Change the last part of the warning to "... please use HZN_NODE_ID in the future."